File: chat2db-server/chat2db-server-domain/chat2db-server-domain-api/src/main/java/ai/chat2db/server/domain/api/model/AiConversation.java
package ai.chat2db.server.domain.api.model;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * AI 会话 DTO
 */
@Data
public class AiConversation {

    private Long id;
    private LocalDateTime gmtCreate;
    private LocalDateTime gmtModified;

    private String conversationId;
    private Long userId;
    private String title;
    private Long dataSourceId;
    private String dataSourceName;
    private String databaseName;
    private String schemaName;
    private Integer messageCount;
    private String lastMessagePreview;
    private String status;
}

File: chat2db-client/src/typings/aiConversation.ts
export interface IAiConversation {
  conversationId: string;
  title?: string | null;
  dataSourceId?: number | null;
  dataSourceName?: string | null;
  databaseName?: string | null;
  schemaName?: string | null;
  messageCount: number;
  lastMessagePreview?: string | null;
  status: 'ACTIVE' | 'ARCHIVED' | 'DELETED';
  gmtCreate?: string;
  gmtModified?: string;
}

export interface IAiMessage {
  messageId: string;
  role: 'user' | 'assistant';
  content: string;
  thinking?: string | null;
  sequenceNo: number;
  gmtCreate?: string;
}

export interface IAiConversationDetail {
  conversation: IAiConversation;
  messages: IAiMessage[];
}

export interface IAiConversationCreateRequest {
  dataSourceId?: number | null;
  databaseName?: string | null;
  schemaName?: string | null;
  initialMessage?: string | null;
}

export interface IAiConversationQueryRequest {
  pageNo: number;
  pageSize: number;
  searchKey?: string;
}

export interface IAiConversationRenameRequest {
  title: string;
}

export interface IPageResponse<T> {
  data: T[];
  pageNo: number;
  pageSize: number;
  total: number;
  hasNextPage?: boolean;
}
